classes/publication/DAO.inc.php

```diff
@@ -386,19 +386,31 @@ public function deleteAllPubIds($contextId, $pubIdType)
      */
     protected function setAuthors(Publication $publication)
     {
-        $publication->setData('authors', iterator_to_array(
-            Services::get('author')->getMany(['publicationIds' => $publication->getId()])
-        ));
+        $publication->setData(
```

Comment:
@NateWr, how do you feel about keeping data fetched from dependent entities in collections vs. arrays? Downstream this leads to changes like https://github.com/pkp/pkp-lib/pull/7211/files#diff-a621a30d446415c7342cc5f53457fb1d44217ad4ff996c419e86adbba63659eeR218. In the back of my head I think simplifying it into an array will probably perform a little better and behave consistently with our current approach, but it might be that collection behaviour becomes useful later -- lazy fetching?

Reply:
I think we should keep such data as a Collection. This will provide consistency when working with a collection of objects (whether fetched from the Repo or attached to another object), take advantage of the `LazyCollection`'s ability to not fetch data that's never used (which is unfortunately common for submission objects), and give us access to collection helper methods.

For example, our schema maps expect to receive a `Collection`. When we map dependent objects, we pass the `Collection` into that entity's schema map. If we converted to array here, we'd have to convert back to a `Collection` when mapping it.

As far as I know, `Collection`s work like arrays for the key use cases, like `foreach`. We just need to be careful about things like `empty`, `$array[1]`, and the `array_map`, `array_filter`, `array_reduce` method.

Reply:
:+1: OK, let's go with Collections!